src/zenml/steps/base_step.py

```diff
@@ -871,33 +873,47 @@ def _apply_configuration(
         logger.debug(self._configuration)
 
     def _validate_configuration(
-        self, config: "StepConfigurationUpdate"
+        self,
+        config: "StepConfigurationUpdate",
+        runtime_parameters: Dict[str, Any],
     ) -> None:
         """Validates a configuration update.
 
         Args:
             config: The configuration update to validate.
+            runtime_parameters: Dictionary of parameters passed to a step from runtime
         """
         settings_utils.validate_setting_keys(list(config.settings))
-        self._validate_function_parameters(parameters=config.parameters)
+        self._validate_function_parameters(
+            parameters=config.parameters, runtime_parameters=runtime_parameters
+        )
         self._validate_outputs(outputs=config.outputs)
 
     def _validate_function_parameters(
-        self, parameters: Dict[str, Any]
+        self,
+        parameters: Dict[str, Any],
+        runtime_parameters: Dict[str, Any],
     ) -> None:
         """Validates step function parameters.
 
         Args:
             parameters: The parameters to validate.
+            runtime_parameters: Dictionary of parameters passed to a step from runtime
 
         Raises:
             StepInterfaceError: If the step requires no function parameters but
                 parameters were configured.
+            RuntimeError: If the step has parameters configured differently in
+                configuration file and code.
         """
         if not parameters:
             return
 
+        conflicting_parameters = {}
         for key, value in parameters.items():
+            if runtime_value := runtime_parameters.get(key, None):
```

Comment:
Same as for the pipeline, `None`/`False`/`0`/... as parameter values will probably not be included in these checks.

Reply:
Fixed